light-client/src/operations/voting_power.rs

```diff
@@ -50,36 +53,104 @@ impl VotingPowerCalculator for ProdVotingPowerCalculator {
     fn voting_power_in(
         &self,
         signed_header: &SignedHeader,
-        validators: &ValidatorSet,
+        validator_set: &ValidatorSet,
     ) -> Result<u64, BoxError> {
+        let signatures = &signed_header.commit.signatures;
+        let validators = validator_set.validators();
+
+        // ensure!(
+        //     validators.len() == signatures.len(),
+        //     // TODO: Raise error
+        // );
+
         // NOTE: We don't know the validators that committed this block,
         //       so we have to check for each vote if its validator is already known.
-        let mut signed_power = 0_u64;
+        let voting_power_needed = self.total_power_of(validator_set) * 2 / 3;
+        let mut tallied_voting_power = 0_u64;
+
+        for (idx, signature) in signatures.into_iter().enumerate() {
+            if signature.is_absent() {
+                continue; // OK, some signatures can be absent.
+            }
+
+            // The vals and commit have a 1-to-1 correspondance (see check above).
+            // This means we don't need the validator address or to do any lookup.
+            let val = validators[idx];
 
-        for vote in &signed_header.signed_votes() {
```

Comment:
Thanks for doing this @romac 🙇 Can't we then delete the `signed_votes()` method?

Reply:
I guess we could, but since this is very much a WIP I didn't remove any code yet.

Reply:
Will do this as part of #342.